tools/build.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+
+import argparse
+from setup import get_build_dir
+import subprocess
+import os
+import time
+
+project_root = get_build_dir() / '..'
+
+def touch_cmake_lists():
+    cmake_lists_path =  project_root / 'CMakeLists.txt'
+    os.utime(cmake_lists_path, times=None)
+
+build_timestamp_path = project_root / '.build_timestamp'
+build_sources_path = project_root / '.build_sources'
+
+def write_build_timestamp(timestamp):
+    with open(build_timestamp_path, 'w') as f:
+        f.write(str(int(timestamp)))
+
+def read_build_timestamp():
+    try:
+        with open(build_timestamp_path, 'r') as f:
+            timestamp = int(f.read().strip())
+            return timestamp
+    except (IOError, ValueError):
+        return 0
```

Comment:
Better: `return -1`, which will *definitely* be smaller than any existing timestamp (and is also a better indicator for errors)